Custom http client enhancement

akrasnoshchok opened the pull request:
#19

ciaran16 commented:
Hey @akrasnoshchok, thanks for opening another PR 💪 

From looking at the [okhttp documentation](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/), it seems like we should only be creating one `OkHttpClient` instance:

> OkHttp performs best when you create a single OkHttpClient instance and reuse it for all of your HTTP calls.

But you can [share an instance](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#customize-your-client-with-newbuilder) and configure it differently, although I'm not sure of the best way of exposing that functionality.

> You can customize a shared OkHttpClient instance with newBuilder. This builds a client that shares the same connection pool, thread pools, and configuration.

I'm also wary of tying this library to a specific OkHttp version or even http library. Perhaps a better approach here would be to add methods to the Onfido builder that wrap OkHttp functionality to support things like logging, configuring timeouts etc.

akrasnoshchok commented:
Hey @ciaran16, thanks for reviewing my PR.

I agree with you about shared OkHttp instance, added newBuilder approach and added clientInterceptor attribute to Onfido builder, check it, please.

I am not sure you should worry about tying onfido library to a specific OkHttp version or http library, if you will decide to migrate to some other http library - you can easily update configuration attributes as well.

akrasnoshchok commented:
Hey @ciaran16, is it ok?
